### PR TITLE
Topic histogram by s4443048

### PR DESCRIPTION
**Pull request by craigdeng**
Histogram of image implemented for Tensorflow.
This function is rewritten and maintained by Puyang Deng. StudentID: s4443048.

### CODE REVIEW

File: image/histogram_s4443048_Puyang_Deng/histogram.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        raise ValueError('Incorrect value for `source_range` argument: {}'.format(source_range))
+    
+    if source_range == 'image':
+        image_min = int(image.min().astype(tf.int64))
+        image_max = int(image.max().astype(tf.int64))
+        
+    elif source_range == 'dtype':
+        image_min, image_max = dtype_limits(image, clip_negative=False)
+        
+    image, offset = _offset_array(image, image_min, image_max)
```
Comment (shakes76) [inline image/histogram_s4443048_Puyang_Deng/histogram.py]: Good to add inline comments even if original code didn't have them

File: image/histogram_s4443048_Puyang_Deng/histogram.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
+import tensorflow as tf
```
Comment (shakes76) [inline image/histogram_s4443048_Puyang_Deng/histogram.py]: Always good to add a header block. Block would include authorship, license, copyright, dates and description of what the driver/module does.

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback  required. Remove unnecessary files**
Request Description minimal. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

File: image/histogram_s4443048_Puyang_Deng/histogram.py (hunk 1 above)
Comment (craigdeng) [reply image/histogram_s4443048_Puyang_Deng/histogram.py]: Sorry for the delay sir! Just add the inline comment

File: image/histogram_s4443048_Puyang_Deng/histogram.py (hunk 2 above)
Comment (craigdeng) [reply image/histogram_s4443048_Puyang_Deng/histogram.py]: Added the header. Sorry for the delay!

**Comment by craigdeng** [comment]
Just added the inline comment and the header block.

**Comment by craigdeng** [comment]
Delete the unnecessary file as well, do I need to create a new pull request?

**Comment by craigdeng** [comment]
Hi Professor Chandra,

Sorry for the delay!
I have just update the code and commit through git, hope you can see it.

Have a nice day!

Kind Regards,
Puyang Deng

在 2019年11月4日，上午11:29，Shakes <notifications@github.com<mailto:notifications@github.com>> 写道：

TF Usage

OK

Good Practice (Design/Commenting)

Good design and comments

Algorithm

Driver Script present
Shows Usage & Demo & Visualisation & Data usage
Module present

Commit Log

Meaningful commit messages
Progressive commits used

Documentation

Description and Comments good
Markdown used PDF submitted

Pull Request

Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch)
Feedback required. Remove unnecessary files
Request Description minimal. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Pull request merged**